python/lsst/ts/MTAlignment/alignment_model.py

```diff
@@ -279,164 +284,201 @@ async def check_status(self):
+            current working frame.
 
-                Returns
-                -------
-                position : str
-                    Position of target, as a point coordinate string,
-                    relative to the current working frame.
+        Raises
+        ------
+        RuntimeError
+            If fail to parse response from controller.
         """
-        return await self.send_command(f"?POS {target}")
+        target_position_response = await self.send_command(f"?POS {target}")
+
+        target_position = parse_offsets(target_position_response)
+
+        if target_position is None:
+            raise RuntimeError(f"Could not parse response: {target_position_response}.")
+        else:
+            return target_position
 
-    async def get_point_position(self, pointgroup, point, collection="A"):
+    async def get_point_position(
+        self, pointgroup: str, point: str, collection: str = "A"
+    ) -> str:
         """Get the position of a previously measured point.
 
         Parameters
         ----------
-        pointgroup : String
+        pointgroup : `str`
             Name of pointgroup the point is in.
-        point : String
+        point : `str`
             Name of point.
-        collection : String
+        collection : `str`
             Name of collection the point is in. Default "A"
 
         Returns
         -------
-        position : str
+        position : `str`
             Point coordinate string
         """
         return await self.send_command(f"?POINT_POS:{collection};{pointgroup};{point}")
 
-    async def get_target_offset(self, target, reference_pointgroup=None):
+    async def get_target_offset(
+        self, target: str, reference_pointgroup: None | str = None
+    ) -> dict[str, str | float]:
         """Get the offset of a target from nominal.
 
         Parameters
         ----------
-        target : str
+        target : `str`
             Target name, e.g. "M1".
         reference_pointgroup : `None` | `str`
             Name of pointgroup that will be used as the frame of reference for
             the offset. If None then use target.
 
         Returns
         -------
-        position : str
-            Offset of target.
+        target_offset : `dict` [`str`, `str` | `float`]
+            Target offset information.
+
+        Raise
+        -----
+        RuntimeError
+            If failed to parse response from controller.
         """
         if reference_pointgroup is None:
             reference_pointgroup = target
-        return await self.send_command(f"?OFFSET:{reference_pointgroup};{target}")
+
+        target_offset_response = await self.send_command(
+            f"?OFFSET:{reference_pointgroup};{target}"
+        )
+
+        target_offset = parse_offsets(target_offset_response)
+
+        if target_offset is None:
+            raise RuntimeError(f"Failed to parse response: {target_offset_response}")
+        else:
+            return target_offset
```

Comment:
Thanks for moving the parsing here.

However, I strongly suggest making `parse_offset` (and your other parsers, such as `parse_single_point_measurement`) raise an informative error if something goes wrong. That way all code that calls your parsers can simply do this:
```
return parse_offsets(target_offset_response)
```
Even if you only call a parser in one place, it arguably puts the responsibility for good error reporting where it belongs.